osclib/stagingapi.py

```diff
@@ -1570,7 +1572,40 @@ def _candidate_adi_project(self):
             adi_index = i + 2
         return self.adi_prj_from_number(adi_index)
 
-    def create_adi_project(self, name):
+    def update_adi_frozenlinks(self, name, src_prj):
+        search = {
+            'package': "@project='%s' and devel/@project='%s'" % (self.project, src_prj),
+        }
+        try:
+            collection = osc.core.search(self.project, **search)['package']
+        except urllib2.HTTPError, e:
+            raise e
+
+        # all packages had matched devel project defined
+        pkglist = [p.attrib['name'] for p in collection.findall('package')]
+
+        flink = ET.Element('frozenlinks')
+        fl_prj = ET.SubElement(flink, 'frozenlink', {'project': self.project})
+
+        project_sourceinfo = ET.fromstring(osc.core.show_project_sourceinfo(self.apiurl, self.project, True))
+        for si in project_sourceinfo.findall('sourceinfo'):
+            pkg = si.get('package')
+            if pkg in pkglist:
+                ET.SubElement(fl_prj, 'package', {'name': pkg, 'srcmd5': si.get('srcmd5'), 'vrev': si.get('vrev')})
+            # check multiple spec ie. sub-package
+            for linked in si.findall('linked'):
+                if linked.get('package') in pkglist:
+                    ET.SubElement(fl_prj, 'package', {'name': pkg, 'srcmd5': si.get('lsrcmd5'), 'vrev': si.get('vrev')})
+
+        # commit frozenlinks
+        url = self.makeurl(['source', name, '_project', '_frozenlinks'], {'meta': '1'})
+        l = ET.tostring(flink)
+        try:
+            http_PUT(url, data=l)
+        except urllib2.HTTPError, e:
+            raise e
```

Comment:
Same as above. Any reason to catch the exception if only raised when caught, otherwise drop try/except?